src/lib/ab.ts

```diff
@@ -30,9 +30,9 @@ export const selectVariant = (test: Test, mvtId: number): Variant => {
             return control;
         } else {
             const otherVariants = test.variants.filter(v => v.name.toLowerCase() !== 'control');
-            return otherVariants[mvtId % otherVariants.length];
+            return otherVariants[mvtId % otherVariants.length] as Variant;
```

Comment:
There is a potential issue here: if a user has set the control proportion but not created any variants then `otherVariants.length ` = 0 and `mvtId % 0` = NaN, which means we will actually return `undefined` here. It's probably best to ensure that in the tooling though, as that shouldn't be a valid test

Reply:
That's a good point. I've just checked and it is possible to have `controlProportionSettings` and only a control, but you're right that this needs to be fixed in the tool